pkg/devserver/lifecycle.go

```diff
@@ -45,3 +53,60 @@ func (l lifecycle) OnFunctionScheduled(
 		Cron:         event.Cron,
 	})
 }
+
+func (l lifecycle) OnFunctionFinished(
+	ctx context.Context,
+	id state.Identifier,
+	item queue.Item,
+	resp state.DriverResponse,
+	s state.State,
+) {
+	now := time.Now()
+	data := map[string]interface{}{
+		"function_id": s.Function().Slug,
+		"run_id":      id.RunID.String(),
+	}
+
+	origEvt := s.Event()
+
+	if dataMap, ok := origEvt["data"].(map[string]interface{}); ok {
+		if inngestObj, ok := dataMap[consts.InngestEventDataPrefix].(map[string]interface{}); ok {
+
+			if dataValue, ok := inngestObj[consts.InvokeCorrelationId].(string); ok {
+				logger.From(ctx).Debug().Str("data_value_str", dataValue).Msg("data_value")
+				data[consts.InvokeCorrelationId] = dataValue
+			}
+		}
+	}
+
+	if resp.Err != nil {
+		data["error"] = resp.UserError()
+	} else {
+		data["result"] = resp.Output
+	}
+
+	evt := event.Event{
+		ID:        ulid.MustNew(uint64(now.UnixMilli()), rand.Reader).String(),
+		Name:      event.FnCompletedName,
+		Timestamp: now.UnixMilli(),
+		Data:      data,
+	}
```

Comment:
Should this use the same `user` as the original event? To get a trail of all events stemming from a particular user this seems reasonable, though this isn't _triggered_ by the user.

Reply:
This is a really good question.  To be honest, I quite like the idea of being able to `POST` the entire event, excluding name:
```
{
  id: "whatever", // for idempotency,
  data: {},
  user: {},
}
```

WDYT?  We might want to be able to invoke with idempotency.  User could be a good idea to add for our future plans.  I don't want to complicate the SDK's API, though.

Reply:
We have support for `data`, `user`, and `v` right now, but `id` is omitted from any event sending in TS. Sounds like we can add that to all entry points (`.invoke()`, `.send()`, `.sendEvent()`) in a follow-up.